[refactor] rentlist 도메인의 .java 파일을 .kt 로 전환

MEOHIN opened the pull request:
## 💡 변경 사항

rentlist 도메인의 .java 파일을 .kt 로 전환

---

### ✅ 특이 사항

- 효상님께서 구현하신 RentListPublicController.java를 제외한 나머지 파일들은 마이그레이션했습니다.
- rentlistService, rentlistController, rentlistReposity를 마이그레이션 하는 과정에서 효상님께서 작성한 decideRentRequest 부분만 제외할 수는 없어서 함께 전환했습니다.
    - 이 과정에서 RentRequestDecisionDto.kt 까지 코틀린으로 전환하였습니다.
    - 이 부분들은 RentListPublicController를 전환하시면서 한 번 더 확인 부탁드립니다.

---

### 🔗 관련 이슈

closed #55

MEOHIN committed:
Refactor: RentList 엔티티 리팩토링 및 개선

- borrowerUser와 rent 필드의 Nullable 속성을 제거하고 lateinit으로 변경
- loanDate와 returnDate 필드의 Nullable 속성 제거

MEOHIN committed:
.java을(를) .kt(으)로 이름 변경

MEOHIN committed:
refactor: RentRequestStatus를 .java > .kt 전환

MEOHIN committed:
.java을(를) .kt(으)로 이름 변경

MEOHIN committed:
refactor: RentListCreateRequestDto를 .java > .kt 전환

MEOHIN committed:
.java을(를) .kt(으)로 이름 변경

MEOHIN committed:
refactor: RentListResponseDto를 .java > .kt 전환

MEOHIN committed:
.java을(를) .kt(으)로 이름 변경

MEOHIN committed:
refactor: RentListRepository를 .java > .kt 전환

MEOHIN committed:
.java을(를) .kt(으)로 이름 변경

MEOHIN committed:
refactor: RentRequestDecisionDto를 .java > .kt 전환

MEOHIN committed:
.java을(를) .kt(으)로 이름 변경

MEOHIN committed:
refactor: RentListService를 .java > .kt 전환

MEOHIN committed:
.java을(를) .kt(으)로 이름 변경

MEOHIN committed:
refactor: RentListController를 .java > .kt 전환

MEOHIN committed:
refactor: RentListController에서 RsData.of 대신 RsData 생성자를 사용하도록 모든 부분을 수정

MEOHIN committed:
refactor: RentListController에서 반환 타입 통일

- returnBook() 메서드의 반환 타입 RsData<Void?>를 RsData<String?>으로 변경
- 불필요한 null처리 제거

MEOHIN committed:
refactor: RentListService를 ServiceException 기반으로 수정

- IllegalArgumentException → ServiceException으로 변경
- RuntimeException → ServiceException으로 변경
- 에러 코드 체계화 (404-1, 400-1, 403-1 등)
- require() → if 조건문으로 변경하여 명시적 예외 처리

MEOHIN committed:
refactor: RentListController를 ServiceException 기반으로 수정

- catch 블록에서 ServiceException 처리 추가
- e.rsData를 활용하여 에러 코드와 메시지 전달

Gooraeng commented on backend/src/main/java/com/bookbook/domain/rentList/controller/RentListController.kt:
아래의 메소드 세 개는 data가 null 인 것 같아요 그래서 타입 Void쪽으로 수정해주시면 좋아보일 것 같습니다
* createRentList
* decideRentRequest
* returnBook

getRentListByUserId 메소드는 비어있어도 list 를 리턴하니까 그 부분도 살펴봐주실 수 있을까요? 🙏

MEOHIN committed:
Refactor: RentListController의 반환 타입 및 예외 처리 개선

- createRentList(), decideRentRequest(), returnBook() 메서드의 반환 타입을 RsData<Void?>로 통일
- ServiceException 처리 방식을 개선하여 공통 RsData 형식을 따르도록 수정
- getRentListByUserId() 메서드의 반환 타입에서 불필요한 nullable 속성 제거

HeegangBae reviewed (approved):
고생하셨습니다.
